nomos-utils/src/fisheryates.rs

```diff
@@ -4,12 +4,24 @@ use rand::{Rng as _, RngCore};
 pub struct FisherYates;
 
 impl FisherYates {
-    pub fn shuffle<R: RngCore, T: Clone>(elements: &mut [T], rng: &mut R) {
+    pub fn shuffle<R: RngCore, T>(elements: &mut [T], rng: &mut R) {
         // Implementation of fisher yates shuffling
         // https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle
         for i in (1..elements.len()).rev() {
             let j = rng.random_range(0..=i);
             elements.swap(i, j);
         }
     }
+
+    pub fn sample<R: RngCore, T: Ord>(
+        elements: impl IntoIterator<Item = T>,
+        sample_size: usize,
+        rng: &mut R,
+    ) -> impl Iterator<Item = T> {
+        let mut elements: Vec<T> = elements.into_iter().collect();
+        elements.sort();
```

Comment:
Is this for determinism?  Why do we need to sort before shuffle?

Reply:
I am not 100% sure but if shuffle is rng then we probably dont need it

Reply:
I defined it like that in the [python spec](https://github.com/logos-co/nomos-specs/blob/c763d473399204b4672deecaf97d81ad533002f8/da/assignations/refill.py#L58) as well. I think is better to sort so its independent of the input order what to chose. This is pseudo-randomness here. But most important is reproducibility. It could be tricky if you pass different collections here, with the same rng and they will return different results. We do not want this (it could also happen depending implementation). So we chose to be safe and sort.